pkg/services/authn/authnimpl/sync/oauth_token_sync.go

```diff
@@ -100,27 +103,49 @@ func (s *OAuthTokenSync) SyncOauthTokenHook(ctx context.Context, identity *authn
 		s.cache.Set(identity.ID, struct{}{}, getOAuthTokenCacheTTL(accessTokenExpires, idTokenExpires))
 		return nil
 	}
-	// FIXME: Consider using context.WithoutCancel instead of context.Background after Go 1.21 update
-	updateCtx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
-	defer cancel()
 
-	if err := s.service.TryTokenRefresh(updateCtx, token); err != nil {
-		if errors.Is(err, context.Canceled) {
-			return nil
-		}
-		if !errors.Is(err, oauthtoken.ErrNoRefreshTokenFound) {
-			s.log.Error("Failed to refresh OAuth access token", "id", identity.ID, "error", err)
-		}
+	lockKey := fmt.Sprintf("oauth-token-sync-%s", identity.ID)
```

Comment:
NIT: identity.ID should be unique here so we don't have to prefix the key

Reply:
Changed, thanks!